sdk/servicebus/Microsoft.Azure.ServiceBus/tests/SenderReceiverClientTestBase.cs

```diff
@@ -275,6 +275,103 @@ internal async Task CancelScheduledMessagesAsyncTestCase(IMessageSender messageS
+                                await messageReceiver.CompleteAsync(message.SystemProperties.LockToken);
+                            }
+
+                            Interlocked.Increment(ref count);
+                        },
+                        new MessageHandlerOptions(ExceptionReceivedHandler) { MaxConcurrentCalls = maxConcurrentCalls, AutoComplete = autoComplete });
+
+                    // Wait for the OnMessage Tasks to finish
+                    var stopwatch = Stopwatch.StartNew();
+                    while (stopwatch.Elapsed.TotalSeconds <= 30)
+                    {
+                        if (count == messageCount)
+                        {
+                            TestUtility.Log($"Too many ('{maxConcurrentCalls}') messages received.");
+                            break;
+                        }
+                        await Task.Delay(TimeSpan.FromSeconds(5));
+                    }
+
+                    Assert.True(count < messageCount);
+                }
+
+        internal async Task MultipleMessageHandlerAsyncTestCase(
+            IMessageSender messageSender,
+            IMessageReceiver messageReceiver,
+            int maxConcurrentCalls,
+            bool autoComplete,
+            int messageCount)
+            {
+            var count = 0;
+
+            void HandlerRegistration(bool unregisterImmediately)
+            {
+                messageReceiver.RegisterMessageHandler(
+                    async (message, token) =>
+                    {
+                        TestUtility.Log($"Received message: SequenceNumber: {message.SystemProperties.SequenceNumber}");
+
+                        if (unregisterImmediately)
+                        {
+                            messageReceiver.UnregisterMessageHandler();
+                            
+                            // recursively register a new handler and expect it to handle the open messages 
+                            HandlerRegistration(false);
+                        }
+
+                        if (messageReceiver.ReceiveMode == ReceiveMode.PeekLock && !autoComplete)
+                        {
+                                                    // Completion should still work, even after message handler is stopped
+                                                    await messageReceiver.CompleteAsync(message.SystemProperties.LockToken);
+                        }
+
+                        Interlocked.Increment(ref count);
+                    },
+                    new MessageHandlerOptions(ExceptionReceivedHandler) { MaxConcurrentCalls = maxConcurrentCalls, AutoComplete = autoComplete });
+            }
+
+            await TestUtility.SendMessagesAsync(messageSender, messageCount);
+            HandlerRegistration(true);
+
+            // Wait for the OnMessage Tasks to finish
+            var stopwatch = Stopwatch.StartNew();
+            while (stopwatch.Elapsed.TotalSeconds <= 60)
```

Comment:
>60 [](start = 53, length = 2)

Again, could you reduce this further (and enable prefetch on the receiver so that messages are received faster)